aspnetcore/fundamentals/middleware/extensibility/sample/Startup.cs

```diff
@@ -0,0 +1,44 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using MiddlewareExtensibilitySample.Data;
+using MiddlewareExtensibilitySample.Middleware;
+
+namespace MiddlewareExtensibilitySample
+{
+    public class Startup
+    {
+        #region snippet1
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<AppDbContext>(options =>
+                options.UseInMemoryDatabase("InMemoryDb"));
+
+            services.AddScoped<IMiddlewareMiddleware>();
```

Comment:
Lets change this to AddTransient (AddScoped isn't incorrect). AddTransient is more efficient since the only thing resolving IMiddlewareMiddleware is the factory.